### ISSUE TITLE
CocoaPods could not find compatible versions for pod "OneSignal"

### ISSUE THREAD

**Issue by cfmitrah** [OneSignal/react-native-onesignal#693]
**Description:**

In android the integration was successful. however in iOS after adding below line in podfile
pod 'react-native-onesignal',
  :path => "../node_modules/react-native-onesignal/react-native-onesignal.podspec",
  :inhibit_warnings => true

I am getting this error after pod install
[!] CocoaPods could not find compatible versions for pod "OneSignal":
  In Podfile:
    react-native-onesignal (from `../node_modules/react-native-onesignal/react-native-onesignal.podspec`) was resolved to 3.2.12, which depends on
      OneSignal (= 2.9.4)

None of your spec sources contain a spec satisfying the dependency: `OneSignal (= 2.9.4)`.

**Environment**

1. "react-native-onesignal": "^3.2.12"
2. added it using npm and linking it with cocoapods

**Steps to Reproduce Issue:**

1.npm install --save react-native-onesignal
2. add this line to pod file 
  pod 'react-native-onesignal',
  :path => "../node_modules/react-native-onesignal/react-native-onesignal.podspec",
  :inhibit_warnings => true
3. run 'pod install'

**Anything else:**

(crash stacktraces, as well as any other information here)

**Comment by Nightsd01** [comment]
@cfmitrah Please try to `cd` into the `/ios` directory and run `pod repo update`.

If that does not work, please post your Podfile

**Comment by Nightsd01** [comment]
Closing due to no response

**Comment by jake-tobin** [comment]
for any fellow travelers,
-->> https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/55#issuecomment-459369818

**Comment by ldco2016** [comment]
@Nightsd01 , 

I am getting this error:

```
[!] CocoaPods could not find compatible versions for pod "OneSignal":
 In Podfile:
 react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 3.4.2, which depends on
 OneSignal (= 2.11.2)
 
/bin/bash failed with return code: 31
return code: 1
None of your spec sources contain a spec satisfying the dependency: `OneSignal (= 2.11.2)`.
 
You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
 
Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.
```

When I run the following commands in Azure DevOps:

`cd ios && rm Podfile.lock && pod deintegrate && cd .. && rm -rf node_modules && rm package-lock.json && npm install && cd ios/ && pod install && pod repo update`

Is this too aggressive? Do I not need to run `rm Podfile.lock && pod deintegrate`? Would `pod install` be sufficient for a CI build tool configuration?
